Rubberduck.Parsing/VBA/AttributesUpdater.cs

```diff
@@ -0,0 +1,176 @@
+            {
+                if (values.Count != 2)
+                {
+                    return;
+                }
+                //Is the key already defined as external key?
+                if(declaration.Attributes.Any(attrbt => attrbt.Name.Equals(attribute, StringComparison.OrdinalIgnoreCase)
+                                                        && attrbt.Values[0].Equals(values[0])))
+                {
+                    return;
+                }
+            }
+            else if (declaration.Attributes.HasAttribute(attribute))
+            {
+                return;
+            }
+
+            var rewriter = rewriteSession.CheckOutModuleRewriter(declaration.QualifiedModuleName);
+            if (declaration.DeclarationType.HasFlag(DeclarationType.Module))
+            {
+                var moduleParseTree = (ParserRuleContext)_parseTreeProvider.GetParseTree(declaration.QualifiedModuleName, CodeKind.AttributesCode);
+                var lastModuleAttribute = moduleParseTree.GetDescendents<VBAParser.ModuleAttributesContext>()
+                    .Where(moduleAttributes => moduleAttributes.attributeStmt() != null)
+                    .SelectMany(moduleAttributes => moduleAttributes.attributeStmt())
+                    .OrderBy(moduleAttribute => moduleAttribute.stop.TokenIndex)
+                    .LastOrDefault();
+                if (lastModuleAttribute == null)
+                {
+                    //This should never happen for a real module.
+                    var codeToInsert = $"Attribute {attribute} ={AttributeValuesText(values)}{Environment.NewLine}";
+                    rewriter.InsertBefore(0, codeToInsert);
+                }
+                else
+                {
+                    var codeToInsert = $"{Environment.NewLine}Attribute {attribute} ={AttributeValuesText(values)}";
+                    rewriter.InsertAfter(lastModuleAttribute.stop.TokenIndex, codeToInsert);
+                }  
+            }
+            else
+            {
+                ParserRuleContext attributesContext = declaration.AttributesPassContext;
+                var firstEndOfLineInMember = attributesContext.GetDescendent<VBAParser.EndOfLineContext>();
+                if (firstEndOfLineInMember == null)
+                {
+                    var codeToInsert = $"{Environment.NewLine}Attribute {attribute} ={AttributeValuesText(values)}";
+                    rewriter.InsertAfter(declaration.AttributesPassContext.Stop.TokenIndex, codeToInsert);
+                }
+                else
+                {
+                    var codeToInsert = $"Attribute {attribute} ={AttributeValuesText(values)}{Environment.NewLine}";
+                    rewriter.InsertAfter(firstEndOfLineInMember.Stop.TokenIndex, codeToInsert);
+                }
+            }
+        }
+
+        private static string AttributeValuesText(IEnumerable<string> attributeValues)
+        {
+            var builder = new StringBuilder();
```

Comment:
Why not just use `$" {string.Join(", ", attributeValues)}`? It should result in roughly the same IL and IMHO is a bit more readable.

Reply:
Because I forgot about `Join`. Will change it.